web/src/settings_org.ts

```diff
@@ -753,25 +763,198 @@ export function deactivate_organization(e: JQuery.Event): void {
     e.preventDefault();
     e.stopPropagation();
 
+    let custom_deletion_time_input = realm.server_min_deactivated_realm_deletion_days;
+    let custom_deletion_time_unit = "days";
+
+    function valid_to(): string {
+        const $deletes_in = $<HTMLSelectOneElement>("select:not([multiple])#deletes_in");
+        const time_input_value = $deletes_in.val()!;
+
+        if (time_input_value === "null") {
+            return $t({defaultMessage: "Data will not be automatically deleted"});
+        }
+
+        let time_in_minutes: number;
+        if (time_input_value === "custom") {
+            if (!util.validate_custom_time_input(custom_deletion_time_input)) {
+                return $t({defaultMessage: "Invalid custom time"});
+            }
+            time_in_minutes = util.get_custom_time_in_minutes(
+                custom_deletion_time_unit,
+                custom_deletion_time_input,
+            );
+        } else {
+            time_in_minutes = Number.parseFloat(time_input_value);
+        }
+
+        if (time_in_minutes === 0) {
+            return $t({defaultMessage: "Data will be deleted immediately"});
+        }
+        // The below is a duplicate of timerender.get_full_datetime, with a different base string.
+        const valid_to = add(new Date(), {minutes: time_in_minutes});
+        const date = timerender.get_localized_date_or_time_for_format(valid_to, "dayofyear_year");
+        const time = timerender.get_localized_date_or_time_for_format(valid_to, "time");
+
+        return $t({defaultMessage: "Data to be deleted on {date} at {time}"}, {date, time});
```

Comment:
It was mentioned in the issue that the UI should be similar to the one we have in Invitation modal:
![image](https://github.com/user-attachments/assets/dbe83233-917c-4e02-abba-ea83c34a1a02)

Just confirm me once, and I will remove the time from this PR.

Reply:
I mean the invitations do stop working on that exact time/date. But for data deletion, I imagine that will be implemented via a daily or weekly cronjob, so I don't think we should be noting that level of detail here. In fact, something like "Data will be deleted after {date}." feels better to me as it's less specific in when it will actually happen.

@alya - Do you have thoughts on this detail?

Reply:
> "Data will be deleted after {date}." feels better to me as it's less specific in when it will actually happen.

Since @alya did not provide further instructions on this topic, I have used the above string to show when the data deletion would take place.